src/main/java/org/aksw/iguana/cc/worker/impl/SPARQLProtocolWorker.java

```diff
@@ -127,21 +129,30 @@ public SPARQLProtocolWorker(long workerId, ResponseBodyProcessor responseBodyPro
      */
     public static void initHttpClient(int threadCount) {
         connectionManager = PoolingAsyncClientConnectionManagerBuilder.create()
-                .setMaxConnTotal(threadCount)
-                .setMaxConnPerRoute(threadCount)
+                .setMaxConnTotal(threadCount * 1000)
+                .setMaxConnPerRoute(threadCount * 1000)
+                .setPoolConcurrencyPolicy(PoolConcurrencyPolicy.LAX)
+                .setConnPoolPolicy(PoolReusePolicy.FIFO)
+                .setDefaultConnectionConfig(org.apache.hc.client5.http.config.ConnectionConfig.custom()
+                        .setConnectTimeout(Timeout.ofSeconds(5))
```

Comment:
Is it in any way important to set these values?

Reply:
The MaxConnTotal and MaxConnPerRoute are important. The rest were kind of left over. They may increase the performance, but I'm not sure.

Reply:
The values for MaxConnTotal and MaxConnPerRoute are kind of arbitrary though. They need to be set to a high value, but I don't know how they correlate to the problems that occured.